core/storage/suggestion/gae_models.py

```diff
@@ -0,0 +1,267 @@
+            reviewer_id=reviewer_id, thread_id=thread_id,
+            assigned_reviewer_id=assigned_reviewer_id, payload=payload).put()
+
+    @classmethod
+    def get_suggestions_by_type(cls, suggestion_type):
+        """Gets all suggestions of a particular type
+
+        Args:
+            suggestion_type: str. The type of the suggestions.
+
+        Returns:
+            list(SuggestionModel). A list of suggestions of the given
+                type, upto a maximum of feconf.DEFAULT_QUERY_LIMIT
+                suggestions.
+        """
+        return cls.get_all().filter(
+            cls.suggestion_type == suggestion_type).fetch(
+                feconf.DEFAULT_QUERY_LIMIT)
+
+    @classmethod
+    def get_suggestions_by_author(cls, author_id):
+        """Gets all suggestions created by the given author.
+
+        Args:
+            author_id: str. The author of the suggestion.
+
+        Returns:
+            list(SuggestionModel). A list of suggestions by the given author,
+            upto a maximum of feconf.DEFAULT_QUERY_LIMIT suggestions.
+        """
+        return cls.get_all().filter(
+            cls.author_id == author_id).fetch(feconf.DEFAULT_QUERY_LIMIT)
+
+    @classmethod
+    def get_suggestions_assigned_to_reviewer(cls, assigned_reviewer_id):
+        """Gets all suggestions assigned to the given user for review.
+
+        Args:
+            assigned_reviewer_id: str. The reviewer assigned to review the
+                suggestion.
+
+        Returns:
+            list(SuggestionModel). A list of suggestions assigned to the given
+                user for review, upto a maximum of feconf.DEFAULT_QUERY_LIMIT
+                suggestions.
+        """
+        return cls.get_all().filter(
+            cls.assigned_reviewer_id == assigned_reviewer_id).fetch(
+                feconf.DEFAULT_QUERY_LIMIT)
+
+    @classmethod
+    def get_suggestions_reviewed_by(cls, reviewer_id):
+        """Gets all suggestions that have been reviewed by the given user.
+
+        Args:
+            reviewer_id: str. The reviewer of the suggestion.
+
+        Returns:
+            list(SuggestionModel). A list of suggestions reviewed by the given
+                user, upto a maximum of feconf.DEFAULT_QUERY_LIMIT
+                suggestions.
+        """
+        return cls.get_all().filter(
+            cls.reviewer_id == reviewer_id).fetch(feconf.DEFAULT_QUERY_LIMIT)
+
+    @classmethod
+    def get_suggestions_by_status(cls, status):
+        """Gets all suggestions with the given status.
+
+        Args:
+            status: str. The status of the suggestion.
+
+        Returns:
+            list(SuggestionModel). A list of suggestions with the given status,
+            upto a maximum of feconf.DEFAULT_QUERY_LIMIT suggestions.
+        """
+        return cls.get_all().filter(
+            cls.status == status).fetch(feconf.DEFAULT_QUERY_LIMIT)
+
+    @classmethod
+    def get_suggestions_by_entity_id(cls, entity_id, entity_type):
```

Comment:
Can entity type go before entity id in the arg list? That's more canonical (first figure out the type of entity, then identify a specific entity of that type).

Reply:
Done